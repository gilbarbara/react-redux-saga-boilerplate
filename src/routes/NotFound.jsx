import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { Container, Heading } from 'styled-minimal';
import Background from 'components/Background';

const StyledContainer = styled(Container)`
  align-items: center;
  text-align: center;

  h1,
  a {
    color: #fff;
    line-height: 1;
  }

  a {
    text-decoration: underline;
  }
`;

const NotFound = () => (
  <Background key="404">
    <StyledContainer layout="fullScreen" verticalPadding>
      <Heading fontSize={100}>404</Heading>
      <Link to="/">
        <Heading as="h2">go home</Heading>
      </Link>
    </StyledContainer>
  </Background>
);

export default NotFound;
