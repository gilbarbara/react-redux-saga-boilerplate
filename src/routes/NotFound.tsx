import { Link } from 'react-router-dom';
import { H2, NonIdealState, Page } from '@gilbarbara/components';

import Background from '~/components/Background';

function NotFound() {
  return (
    <Background key="404" data-component-name="NotFound">
      <Page>
        <NonIdealState />
        <Link to="/">
          <H2>go home</H2>
        </Link>
      </Page>
    </Background>
  );
}

export default NotFound;
