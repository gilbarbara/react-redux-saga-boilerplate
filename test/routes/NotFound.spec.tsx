import React from 'react';
import { fireEvent, navigate, render, screen } from 'test-utils';

import NotFound from '~/routes/NotFound';

describe('NotFound', () => {
  it('should render properly', () => {
    render(<NotFound />);

    expect(screen.getByTestId('NotFound')).toMatchSnapshot();
  });

  it('should redirect to home', () => {
    navigate({ pathname: '/some-page' });

    render(<NotFound />);

    fireEvent.click(screen.getByRole('link'));

    expect(window.location.pathname).toBe('/');
  });
});
