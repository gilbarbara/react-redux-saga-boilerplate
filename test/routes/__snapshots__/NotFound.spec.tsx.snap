// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > should render properly 1`] = `
<div
  class="css-146gbh6"
  data-component-name="NotFound"
>
  <div
    class="css-px2j5d"
    data-component-name="Page"
  >
    <div
      class="css-wc6egs"
      data-component-name="Box"
    >
      <div
        class="css-vacfv5"
        data-component-name="NonIdealState"
      >
        <div
          class="css-16wmnst"
          data-component-name="Box"
        >
          <span
            class="css-1oyk1n8"
            data-component-name="Icon"
            name="smile-none"
          >
            <svg
              height="64"
              src="https://cdn.jsdelivr.net/npm/css.gg/icons/svg/smile-none.svg"
              title="smile-none"
              width="64"
            />
          </span>
        </div>
        <div
          class="css-11oqjv"
          data-component-name="Box"
        >
          <h2
            class="css-wbv4yw"
            data-component-name="H2"
          >
            Page not found
          </h2>
          <p
            class="css-1iqh4jp"
            data-component-name="Paragraph"
          >
            <span
              class="css-lo0v0o"
              data-component-name="Text"
            >
              We are sorry, but the page you requested doesn't exist.
            </span>
          </p>
        </div>
      </div>
      <a
        href="/"
      >
        <h2
          class="css-1bwodyc"
          data-component-name="H2"
        >
          go home
        </h2>
      </a>
    </div>
  </div>
</div>
`;
